sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/repository/support/StringBasedCosmosQuery.java

```diff
@@ -52,24 +52,36 @@ public Object execute(final Object[] parameters) {
         final CosmosParameterAccessor accessor = new CosmosParameterParameterAccessor(getQueryMethod(), parameters);
         final ResultProcessor processor = getQueryMethod().getResultProcessor().withDynamicProjection(accessor);
 
+        /*
+         * The below for loop is used to handle two unique use cases with annotated queries.
+         * Annotated queries are defined as strings so there is no way to know the clauses
+         * being used in advance. Some clauses expect an array and others expect just a list of values.
+         * (1) IN clauses expect the syntax 'IN (a, b, c) which is generated from the if statement.
+         * (2) ARRAY_CONTAINS expects the syntax 'ARRAY_CONTAINS(["a", "b", "c"], table.param) which
+         *     is generated from the else statement.
+         */
         String expandedQuery = query;
         List<SqlParameter> sqlParameters = new ArrayList<>();
         for (int paramIndex = 0; paramIndex < parameters.length; paramIndex++) {
             Parameter queryParam = getQueryMethod().getParameters().getParameter(paramIndex);
-            if (parameters[paramIndex] instanceof Collection) {
-                List<String> expandParam = ((Collection<?>) parameters[paramIndex]).stream()
-                    .map(Object::toString).collect(Collectors.toList());
-                List<String> expandedParamKeys = new ArrayList<>();
-                for (int arrayIndex = 0; arrayIndex < expandParam.size(); arrayIndex++) {
-                    String paramName = "@" + queryParam.getName().orElse("") + arrayIndex;
-                    expandedParamKeys.add(paramName);
-                    sqlParameters.add(new SqlParameter(paramName, toCosmosDbValue(expandParam.get(arrayIndex))));
-                }
-                expandedQuery = expandedQuery.replaceAll("@" + queryParam.getName().orElse(""), String.join(",", expandedParamKeys));
-            } else {
-                if (!Pageable.class.isAssignableFrom(queryParam.getType())
-                    && !Sort.class.isAssignableFrom(queryParam.getType())) {
-                    sqlParameters.add(new SqlParameter("@" + queryParam.getName().orElse(""), toCosmosDbValue(parameters[paramIndex])));
+            String paramName = queryParam.getName().orElse("");
+            if (!paramName.equals("")) {
+                String modifiedExpandedQuery = expandedQuery.toLowerCase().replaceAll("\\s+", "");
```

Comment:
This is still doing the rather expensive transformation for every parameter? My suggestion was to do it outside of the for loop - so make it O(1) instead of O(n) (with n being parameter count)